themes/default/content/blog/replicating-data-to-support-multi-region-applications/index.md

```diff
@@ -0,0 +1,217 @@
+
+Let's look at an example of using Pulumi to define an Azure Cosmos DB account with multi-region replication:
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure-native";
+
+const resourceGroup = new azure.resources.ResourceGroup("myResourceGroup");
+
+const cosmosDbAccount = new azure.documentdb.DatabaseAccount("myCosmosDbAccount", {
+    resourceGroupName: resourceGroup.name,
+    location: resourceGroup.location,
+    locations: [
+        { locationName: "East US" },
+        { locationName: "West US" },
+        // Add additional Azure regions for replication
+    ],
+    databaseAccountOfferType: "Standard",
+    consistencyPolicy: {
+        defaultConsistencyLevel: "Session",
+    },
+});
+```
+
+In this example, the `locations` parameter is used to specify the Azure regions where Cosmos DB replicas will be deployed, enabling global data distribution. Aligning these locations with the locations where your application code is deployed allows you to minimize latency due to reading from the database.
+
+Additionally, you can also enable multi-region writes. This is typically useful for write-active scenarios, ensuring that write operations can be directed to any region with a writable replica, and can reduce latency associated with writes to the database. This reduction in write latency often translates directly into improved performance for a multi-region application.
+
+```typescript
+const cosmosDbAccount = new azure.documentdb.DatabaseAccount("myCosmosDbAccount", {
+    resourceGroupName: resourceGroup.name,
+    … // more code
+    enableMultipleWriteLocations: true,
+});
+```
+
+### Consistency
+
+You will need to choose a consistency level that aligns with your application requirements, such as "Session," "Bounded staleness," "Eventual," "Strong," or "Consistent prefix." [This page](https://learn.microsoft.com/en-us/azure/cosmos-db/consistency-levels) has more details on the different consistency levels, but let’s look at a couple specific configurations.
+
+Recall that strong consistency ensures that all nodes or replicas in the system view the same data simultaneously, regardless of which node they are accessing. In other words, after performing a write operation, any subsequent read action from any node must return the most recent write value. This consistency mode guarantees the linear ordering of processes, and the system behaves like a single, coherent entity. Strong consistency prioritizes consistency over availability in the event of network partition.
+
+You can prioritize strong consistency by configuring the Azure Cosmos DB account to use the "Strong" consistency level. This ensures that all read and write operations observe the most recent version of the data across all replicas.
```

Comment:
May be worth adding a note that this typically hinders performance as the write must be acknowledged by multiple nodes.